capitalone/rubicon-ml
# Adding RubiconJSON docs notebook + API Reference

closes: #315 

---

## What
  * Creates docs notebook for new `RubiconJSON` class
  * Adds notebook to docs in **Tutorials** section
  * Adds docstrings for `RubiconJSON` and `RubiconJSON.search`
  * Adds `RubiconJSON` and `RubiconJSON.search` to **API reference**
  * Adds some type checking that was missing in `RubiconJSON.search` (revealed by error thrown when creating notebook)
  * Lifts `RubiconJSON` to top level `rubicon_ml` in `rubicon_ml/__init__.py`

## How to Test
  * Run Notebook cells
  * Run `python -m pytest tests/unit/client/test_rubicon_json_client.py`
  * Build docs locally and see that notebook is under **Tutorials** and `RubiconJSON` in **API Reference**
       * `conda activate rubicon-ml-docs`
       * `make html`
       * `open ./build/html/index.html`
